apps/backend/src/lib/controller/chip/managed/leaderboard/index.ts

```diff
@@ -35,3 +62,69 @@ ManagedChipClient.prototype.UpdateLeaderboardEntry = async function (
     throw new Error("Failed to update leaderboard entry");
   }
 };
+
+ManagedChipClient.prototype.GetUserLeaderboardPosition = async function (
+  username: string,
+  chipIssuer: ChipIssuer
+): Promise<LeaderboardPosition | null> {
+  try {
+
+    /*
+    1. Get list of entries for chipIssuer, ordered on descending tap count
+    subquery1 = SELECT * FROM "LeaderboardEntry" WHERE "chipIssuer"='TESTING' ORDER BY "tapCount" DESC
+
+    2. Select username and calculated row number from entries
+    subquery2 = SELECT username, row_number() OVER() as "leaderboardPosition" FROM (subquery1) AS sorted_entries
+
+    3. Select row number based on username
+    query = SELECT row_number FROM (subquery2) AS filtered_entries WHERE "username"='pass2';
+
+    Working query:
+    SELECT "leaderboardPosition" FROM (SELECT username, row_number() OVER() as "leaderboardPosition" FROM (SELECT * FROM "LeaderboardEntry" WHERE "chipIssuer"='TESTING' ORDER BY "tapCount" DESC) AS sorted_entries) AS rowed_sorted_entries WHERE "username"='pass2';
+    */
```

Comment:
This query seems too complex. I'll keep mulling over ways to simplify it. If others have ideas I'm all ears. It does work tho.

Reply:
LGTM! Although I would prefer to convert this into pure prisma queries, so that it's easier to read and maintain. I'll add an issue for that in v1.3 milestone

Reply:
Made it #100

Reply:
Sounds good, definitely agreed, when possible I tried to default to prisma, but found it difficult to use for a few of these queries.